packages/core/src/2d/sprite/Sprite.ts

```diff
@@ -0,0 +1,251 @@
+      this._pixelsPerUnit = value;
+      this._setDirtyFlagTrue(Sprite._VERTICES_FLAG);
+    }
+  }
+
+  /**
+   * Constructor a sprite.
+   * @param engine - Engine to which the sprite belongs
+   * @param texture - Texture from which to obtain the sprite
+   * @param rect - Rectangular section of the texture to use for the sprite
+   * @param pivot - Sprite's pivot point relative to its graphic rectangle
+   * @param pixelsPerUnit - The number of pixels in the sprite that correspond to one unit in world space
+   */
+  constructor(
+    engine: Engine,
+    texture: Texture2D,
+    rect: Rect = null,
+    pivot: Vector2 = null,
+    pixelsPerUnit: number = 100
+  ) {
+    super(engine);
+
+    if (!rect) {
+      rect = new Rect(0, 0, texture.width, texture.height);
+    }
+    if (!pivot) {
+      pivot = new Vector2(0.5, 0.5);
+    }
+
+    if (rect.x + rect.width > texture.width || rect.y + rect.height > texture.height) {
+      throw new Error("rect out of range");
+    }
+
+    // Init.
+    this._dirtyFlag = Sprite._VER_UV_TRI_FLAG;
+    this.texture = texture;
+    this.rect = rect;
+    this.textureRect = rect;
+    this.pivot.setValue(pivot.x * rect.width, pivot.y * rect.height);
+    this.pixelsPerUnit = pixelsPerUnit;
+  }
+
+  /**
+   * @override
+   */
+  _onDestroy() {
+    // TODO
+    if (this._texture) {
+      this._texture = null;
+    }
+  }
+
+  /**
+   * Update mesh.
+   */
+  private _updateMesh() {
+    const { _pixelsPerUnit, _pivot, _tempVec2, vertices, uv, triangles } = this;
+
+    // Update vertices.
+    if (this._isContainDirtyFlag(Sprite._VERTICES_FLAG)) {
+      const { width, height } = this._rect;
+
+      // Get the pivot coordinate in 3D space.
+      Vector2.scale(_pivot, 1.0 / _pixelsPerUnit, _tempVec2);
+      // Get the width and height in 3D space.
+      const realWidth = width / _pixelsPerUnit;
+      const realHeight = height / _pixelsPerUnit;
+
+      // Top-left.
+      vertices[0].setValue(-_tempVec2.x, realHeight - _tempVec2.y);
+      // Top-right.
+      vertices[1].setValue(realWidth - _tempVec2.x, realWidth - _tempVec2.y);
+      // Bottom-right.
+      vertices[2].setValue(realWidth - _tempVec2.x, -_tempVec2.y);
+      // Bottom-left.
+      vertices[3].setValue(-_tempVec2.x, -_tempVec2.y);
+    }
+
+    // Update uvs.
+    if (this._isContainDirtyFlag(Sprite._UV_FLAG)) {
+      // Top-left.
+      uv[0].setValue(0, 0);
+      // Top-right.
+      uv[1].setValue(1, 0);
+      // Bottom-right.
+      uv[2].setValue(1, 1);
+      // Bottom-left.
+      uv[3].setValue(0, 1);
+    }
+
+    // Update triangles.
+    if (this._isContainDirtyFlag(Sprite._TRIANGLES_FLAG)) {
+      triangles[0] = 0;
+      triangles[1] = 2;
+      triangles[2] = 1;
+      triangles[3] = 2;
+      triangles[4] = 0;
+      triangles[5] = 3;
+    }
+  }
+
+  /**
+   * Update mesh data of the sprite.
+   * @returns True if the data is refreshed, false otherwise.
+   */
+  updateMeshData(): boolean {
```

Comment:
This is internal function or public function,internal should use _XX and add internal comment

Reply:
> This is internal function or public function,internal should use _XX and add internal comment

done